Source/Turbo/Navigator/NavigationHierarchyController.swift

```diff
@@ -94,22 +94,31 @@ class NavigationHierarchyController {
             navigationController.present(alert, animated: proposal.animated)
         }
     }
+    
+    private var isInModalContext: Bool {
+        navigationController.presentedViewController != nil
+        && !modalNavigationController.isBeingDismissed
+    }
 
     private func navigate(with controller: UIViewController, via proposal: VisitProposal) {
         switch proposal.context {
         case .default:
             if let visitable = controller as? Visitable {
                 delegate.visit(visitable, on: .main, with: proposal.options)
             }
+            let willContextSwitch = isInModalContext
```

Comment:
For consistency in naming, how about `willDismissModalContext`?

Reply:
What about `willReplaceModalContext` and `didReplaceModalContext`? I like that it's related to the `replace` action, and it reads well in this case.

Reply:
Sounds good to me @olivaresf 👍